### PR TITLE
Feature/visibility checks2

### LINKED ISSUE #1624: Visibility of API that is private for a user group
# Definition of done
* [x] If the API is visible to a group, only those people can 
  * [x] see the API in the catalog
  * [x] see the API profile page
* [x] If an unauthorized user has the direct link to the API profile, they must see a 403 .

# Wireframe

1. API Catalog Card View
apicatalogprivate

2. API Catalog table View
apicatalogtableprivate

In both cases, Users having access to Private API should be able to see them in API Catalog.
Respective Card & Table will contain a tag named Private.
Public APIs should not have any tags

3. API profile view
apiprofileprivate

 Tag should appear beside API name.
Should be larger in size than card/table view

### LINKED ISSUE #1686: Administrators cannot see private APIs when not in the managers list
Originally reported by @marla-singer in #1669 

Right now, Apinf admin users **can't see private APIs**

# Screenshots

1. User is Apinf Administrator. Catalog has 4 apis
joxi_screenshot_1475577095313

2. Log in like one of api owner. Set api as private
joxi_screenshot_1475577239491

3. Log in like Apinf administrator. Catalog has 3 apis
joxi_screenshot_1475577184490

### PR DESCRIPTION
**Pull request by jykae**
Closes #1624
Closes #1673
Closes #1686 

Replaces #1697

### CODE REVIEW

**Comment by jykae** [comment]
@marla-singer this replaces #1697 could you review?

**Comment by marla-singer** [comment]
@jykae Check only the tags work or all visibility feature work?

**Comment by jykae** [comment]
@marla-singer my work is checks & tags, but take also overall view of the feature.

File: apis/client/view/header/header.html [hunk 1]
```diff
@@ -8,6 +8,9 @@ <h1 id="api-header">
           {{ api.name }}
           {{> viewApiStatus api=api width="0.4" }}
         </h1>
+        {{#unless api.isPublic }}
```
Comment (marla-singer) [inline apis/client/view/header/header.html]: Please add a space between # and unless

File: catalogue/client/grid/grid.html [hunk 2]
```diff
@@ -7,21 +7,24 @@
...
                class="api-card-name">
               {{ api.name }}
+              {{> viewApiStatus api=api width="0.6" }}
             </a>
             <span class="api-card-author">
               {{_ "catalogueGrid_addedBy" }}
               {{ api.getApiManagersByName }}
             </span>
-          <p class="api-card-description">
+            {{#unless api.isPublic }}
```
Comment (marla-singer) [inline catalogue/client/grid/grid.html]: Please add a space between # and unless

File: catalogue/client/table/table.html [hunk 3]
```diff
@@ -17,15 +17,20 @@ <h4 class="api-catalogue-table-title">
...
-            <i>
-              {{_ "catalogueTable_addedBy" }}
-              {{ api.getApiManagersByName }}
-            </i>
-            <br>
-            <i>
-              {{_ "catalogueTable_added" }}
-              {{ api.relativeCreatedAt }}
-            </i>
+            {{#unless api.isPublic }}
```
Comment (marla-singer) [inline catalogue/client/table/table.html]: Space between # and unless

File: catalogue/client/grid/grid.less [hunk 4]
```diff
@@ -16,7 +16,7 @@
   height: 10.7em;
   margin-bottom: 1.87em;
 
-  .icon-indicator {
+  .api-status-color {
```
Comment (marla-singer) [inline catalogue/client/grid/grid.less]: Why did you change it? Because of it the status indicator has wrong place:
Found:
joxi_screenshot_1475848265426
Expected:
joxi_screenshot_1475848243115
Or add class "pull-right" after variable "className" in this line
https://github.com/apinf/platform/blob/develop/apis/client/view/status/status.js#L23

**Comment by marla-singer** [comment]
Api owner doesn't watch the others public apis. Screenshots:

1. Go to catalog, pick the other public api.
joxi_screenshot_1475848537196

2. Found result: Forbidden
joxi_screenshot_1475848566195


Also it looks like you forgot add i18n tag to json file.

**Comment by marla-singer** [comment]
@jykae As for me the red rectangle part looks very strange. That's not obvious how i can use it?

**Comment by jykae** [comment]
@marla-singer Thanks for review, other things look fine founds, but the last one is not clear. 

Authorized users part is @brylie 's work from another PR. Already accepted.
Only API owners & admins are able to view/edit API when it's private + authorized users can view
It is used to give single users permissions to view API even when API is globally marked Private.
@marla-singer Do you think it would need help text of some sort about usage? I think we can mark that as enhancement task.

**Comment by marla-singer** [comment]
@jykae Yes, I think It should be some message to describe the purpose of this field as sentence you wrote me above "It is used to give single users permissions to view API even when API is globally marked Private." 
Okay, enhancement task sounds good too.

File: catalogue/client/grid/grid.less (hunk 4 above)
Comment (brylie) [inline catalogue/client/grid/grid.less]: I like the `pull-right` approach. Simple, and doesn't require any new CSS/Less.

**Comment by brylie** [comment]
@jykae would you add the help text where @marla-singer suggests? Otherwise, how near is this PR from being merged?

**Comment by jykae** [comment]
@brylie @marla-singer Ok, I add suggested help text. Working on this now.

**Comment by brylie** [comment]
@jykae thanks. What is the estimate of time to complete?

**Comment by jykae** [comment]
@brylie 12.5 mins

**Comment by brylie** [comment]
Perfect!

**Comment by jykae** [comment]
@brylie sorry went bit over estimate, @marla-singer please review changes

**Comment by marla-singer** [comment]
@jykae api-indicator looks bad with class pull-right :worried:
joxi_screenshot_1476105322244

Please remove class 'pull-right' and return '.icon-indicator' class in grid.less 
It's almost ready to merge. Just fix css style

Also these doesn't have any help text. Maybe you add like on screenshot? It's just text in `<p>` tag after `<h3>` tag
joxi_screenshot_1476105692845

**Comment by jykae** [comment]
@marla-singer sorry, I don't know what you mean with "return '.icon-indicator' class in grid.less"

**Comment by marla-singer** [comment]
@jykae It must be so 
https://github.com/apinf/platform/blob/develop/catalogue/client/grid/grid.less#L19

**Comment by jykae** [comment]
@marla-singer ok, done. I don't know how that dropped out.. :)

**Comment by marla-singer** [comment]
Merging

**Comment by bajiat** [comment]
thanks to both @jykae and @marla-singer
